grobid-core/src/main/java/org/grobid/core/data/Date.java

```diff
@@ -144,6 +148,71 @@ public boolean isAmbiguous() {
         return false;
     }
 
+    public static String toISOString(Date date) {
+        int year = date.getYear();
+        int month = date.getMonth();
+        int day = date.getDay();
+
+        String when = "";
+        if (year != -1) {
+            if (year <= 9)
+                when += "000" + year;
+            else if (year <= 99)
+                when += "00" + year;
+            else if (year <= 999)
+                when += "0" + year;
+            else
+                when += year;
+            if (month != -1) {
+                if (month <= 9)
+                    when += "-0" + month;
+                else
+                    when += "-" + month;
+                if (day != -1) {
+                    if (day <= 9)
+                        when += "-0" + day;
+                    else
+                        when += "-" + day;
+                }
+            }
+        }
+        return when;
+    }
+
+    /**
+     * Merge into a given date the date information from a second date. 
+     * The merging follows the year, month, day sequence. If the years
+     * for instance clash, the merging is stopped. 
+     *
+     * Examples of merging: 
+     * "2010" "2010-10" -> "2010-10"
+     * "2010" "2010-10-27" -> "2010-10-27"
+     * "2010-10" "2010-10-27" -> "2010-10-27"
+     * "2010-10-27" "2010-10" -> "2010-10-27"
+     * "2011-10" "2010-10-27" -> "2011-10"
+     * "2010" "2016-10-27" -> "2010"
+     * "2011" "2010" -> 2011
+     */
+    public static Date merge(Date date1, Date date2) {
+        if (date1.getYear() != -1 && date2.getYear() != -1) {
```

Comment:
if date1.getYear() == -1, this still returns date1.
What about something like
```
if (date1.getYear() == -1) {
    return date2;
}
if (date2.getYear() == -1) {
    return date1;
}
```
// rest of the method as-is

Reply:
Indeed, I share the same concern about this. If dat 1 is non existent doesn't make sense to take date2?

Reply:
yes! when I wrote it, I was considering that it is irrelevant to try to merge something when data1 is undefined, but this is not covering the normal general usage.